tools-public/toolpad/resources/queryHeadlessLibrariesDownloads.ts

```diff
@@ -0,0 +1,81 @@
+// Toolpad queries:
+
+const formatDatePart = (datePart) => {
+  return `${datePart < 10 ? "0" : ""}${datePart}`;
+}
+
+const getDateString = (date: Date) => {
+  const month = date.getMonth() + 1;
+  const day = date.getDate();
+  return `${date.getFullYear()}-${month < 10 ? "0" : ""}${month}-${day < 10 ? "0" : ''}${date.getDate()}`
+}
+
+export const getPackages = (inData) => {
+  const packages: string[] = [];
+  Object.keys(inData ?? {}).forEach(packageName => {
+    packages.push(packageName);
+  });
+  return packages;
+}
+
+const getMonthKey = (date: string) => {
+  return date.slice(0, -2) + "01";
+}
+
+export const prepareData = (inData) => {
+  const date = new Date(2022, 6, 1, 0, 0, 0, 0);
+  const today = new Date();
+  const packages = getPackages(inData);
+
+  const monthsData = {};
+
+  while(date < today) {
+    monthsData[getDateString(date)] = {};
+    packages.forEach(packageName => {
+      monthsData[getDateString(date)][packageName] = 0;
+    })
+    date.setMonth(date.getMonth() + 1);
+  }
+
+  packages.forEach(packageName => {
+    Object.keys(inData[packageName]).map(date => {
+      const monthKey = getMonthKey(date);
+      monthsData[monthKey][packageName] += inData[packageName][date];
+    }) 
+  });
+
+  const data: object[] = [];
+
+  Object.keys(monthsData).forEach((date) => {
+    const entry = {
+      date,
+      ...monthsData[date],
+      '@mui/base': monthsData[date]['@mui/base'] +  monthsData[date]['@mui/core'] - monthsData[date]['@mui/material'],
```

Comment:
A side note, in https://docs.google.com/spreadsheets/d/1FHxZ456e1t_MNJZg9wrq7pLRFzknZ6jMBhFuW1j1Q8o/edit#gid=590909841, I also subtract with `@mui/joy`, underestimating the actual downloads of Base UI. Soon or later, we can move to `@mui/core-downloads-tracker`

<img width="1406" alt="Screenshot 2023-05-17 at 19 05 21" src="https://github.com/mui/mui-public/assets/3165635/35b2c75a-04b4-4068-bcfd-6dbd700a9e1a">

Reply:
I will leave it as is for now

Reply:
Yeah, agree, makes more sense.